packages/core/src/application.ts

```diff
@@ -121,15 +135,18 @@ export class Application extends Context {
     request: ServerRequest,
     response: ServerResponse,
   ) {
-    if (request.method === 'GET' && request.url === '/openapi.json') {
+    if (request.method === 'GET' && request.url &&
+      request.url in OPENAPI_SPEC_MAPPING) {
       // NOTE(bajtos) Regular routes are handled through Sequence.
       // IMO, this built-in endpoint should not run through a Sequence,
       // because it's not part of the application API itself.
       // E.g. if the app implements access/audit logs, I don't want
       // this endpoint to trigger a log entry. If the server implements
       // content-negotiation to support XML clients, I don't want the OpenAPI
       // spec to be converted into an XML response.
-      return this._serveOpenApiSpec(request, response);
+      // tslint:disable-next-line:no-any
```

Comment:
Is this tslint comment still needed? AFAICT, `OPENAPI_SPEC_MAPPING` is a well known type `{ [key: string] : OpenApiSpecOptions; }` now, where is `any` coming from?